### ISSUE TITLE
/dev/ttyUSB devices not supported

### ISSUE THREAD

**Issue by dceejay** [dceejay/nodered.snap#6]
hot plug serial devices like USB serial ports are not currently supported by Snapd
This issue just to track that upstream

**Comment by anonymouse64** [comment]
Hotplug support has landed in an experimental state, see https://forum.snapcraft.io/t/hotplug-support/10750

**Comment by dceejay** [comment]
How do we check snapd version currently shipping with apt-get update ?

**Comment by anonymouse64** [comment]
The version of snapd depends on what distro you are running. Some distrubutions such as Ubuntu and Debian support what's called re-exec, where snapd updates itself without going through the distro packages. Other distros such as Fedora disable re-exec and you have to wait for the distro to update snapd.

In any case, you can check the release of snapd installed on your system with:

```
$ snap version
```

**Comment by dceejay** [comment]
great thanks. so Ubuntu 19.04 (or my version) has 2.41 so this support is now "in" ? and not just experimental ?

**Comment by anonymouse64** [comment]
Yes and no. It is available in snapd but is still experimental in 2.41. We are actually waiting for more feedback from users on the feature before taking it out of experimental, so if you could give it a try with your node-red snap that would be great :smile: 

As to how to enable it from experimental, the basic gist is to run:

```
$ sudo snap set core experimental.hotplug=true
$ sudo systemctl restart snapd
```
